test/full-test.js

```diff
@@ -0,0 +1,285 @@
+const chai = require('chai');
+const chaiAsPromised = require('chai-as-promised');
+const Web3 = require('web3');
+const utils = require('./utils');
+const helper = require('./helper');
+
+const ERC20ExampleToken = artifacts.require('ERC20ExampleToken');
+const LiBAStruct = artifacts.require('LiBAStruct');
+const LiBAAsker = artifacts.require('LiBAAsker');
+const LiBABidder = artifacts.require('LiBABidder');
+const LiBA = artifacts.require('LiBA');
+const PoLC = artifacts.require('PoLC');
+
+chai.use(chaiAsPromised);
+const assert = chai.assert;
+
+const AUCTION_DEPOSIT = 1000;
+const MIN_CELER = 10;
+const VALUE = 600;
+const DURATION = 3;
+const MAX_RATE = 10;
+const MIN_VALUE = 2;
+const BID = {
+    rate: 5,
+    value: VALUE / 6,
+    celerValue: 1000,
+    salt: 100,
+    commitmentId: 0
+};
+
+contract(
+    'Comprehensive LiBA test',
+    ([asker, bidder0, bidder1, bidder2, bidder3, bidder4]) => {
+        let celerToken;
+        let liba;
+        let polc;
+        let auctionId;
+        let libaHelper;
+        const bids = {};
+
+        before(async () => {
+            celerToken = await ERC20ExampleToken.new();
+            polc = await PoLC.new(celerToken.address, 100);
+
+            const libaStruct = await LiBAStruct.new();
+            await LiBAAsker.link('LiBAStruct', libaStruct.address);
+            const libaAsker = await LiBAAsker.new();
+            await LiBABidder.link('LiBAStruct', libaStruct.address);
+            const libaBidder = await LiBABidder.new();
+
+            await LiBA.link('LiBAStruct', libaStruct.address);
+            await LiBA.link('LiBAAsker', libaAsker.address);
+            await LiBA.link('LiBABidder', libaBidder.address);
+            liba = await LiBA.new(
+                celerToken.address,
+                polc.address,
+                false,
+                MIN_CELER
+            );
+
+            await polc.setLibaAddress(liba.address);
+            await celerToken.transfer(bidder0, 10000);
+            await celerToken.transfer(bidder1, 10000);
+            await celerToken.transfer(bidder2, 10000);
+            await celerToken.transfer(bidder3, 10000);
+            await celerToken.transfer(bidder4, 10000);
+
+            libaHelper = new helper.LiBAHelper(polc, liba, celerToken);
+            utils.calculateBidHash(BID);
+            for (const bidder of [
+                bidder0,
+                bidder1,
+                bidder2,
+                bidder3,
+                bidder4
+            ]) {
+                bids[bidder] = { ...BID };
+            }
+        });
+
+        it('less bid value than asked auction value', async () => {
```

Comment:
What is the expected result of this test description? Why not use the format used before like "it should do sth when sth happens", which I think is more readable?

Reply:
The test does not do a single thing.